Docs migrates3 dh

dhirsch1 opened the pull request:
Migration of the updated Amazon S3 Connector doc to the Mulesoft site.

dhirsch1 committed:
Entered edits.
Entered edits.

dhirsch1 committed:
Copied files to this branch
Copied files to this branch

dhirsch1 committed:
Fix to link to reference in Design Center topic and wording of Addtional Configuration Information in TOC
Fix to link to reference in Design Center topic and wording of Addtional Configuration Information in TOC

dhirsch1 committed:
Fixed link to reference in Design Center - case issue
Fixed link to reference in Design Center - case issue

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-config-topics.adoc:
```diff
@@ -0,0 +1,44 @@
+
+In the `Create Object` operation, set the `Content Length` to a value greater than zero. If the `Content Length` is set to zero (0), the `Create Object` operation creates a zero-byte object.
+
+== To Use an AWS KMS Master Key
```
Would you be ok dropping "To" from this title and the next?

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-design-center.adoc:
```diff
@@ -0,0 +1,92 @@
+. Exit from *Let's get started* by clicking *Go straight to canvas*.
+. Click on the Trigger card.
+. If you are using the connector as an input source trigger, search for the connector and select an operation. Otherwise, search for and select *HTTP* or *Scheduler*.
+. Complete the fields with the values you want to configure for your trigger.
```
Please provide more details about the fields. No customer benefits from a generic statement like "complete the fields" -- they know that, but they need more info about what info they need to complete the fields. There are most likely similar fields between each operation. Just cover those. This is especially true for this connector with only 2 trigger operations -- if we can't cover 2 operations, what's the point of documenting the use of DC?

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-design-center.adoc:
```diff
@@ -0,0 +1,92 @@
+* *On Deleted Object* - Initiates access to your app when an Amazon S3 object is deleted.
+* *On New Object* - Initiates access to your app when an Amazon S3 object is created.
++
+Alternatively, you can specify the trigger as an HTTP Listener as input to your application from a browser or an application like Postman or curl, or you can use Scheduler to provide input to your application at a specific time.
```
This sentence is missing info. You state that you can specify a trigger as an HTTP Listener, but then the next phrase is unclear if you're describing how the trigger or the Listener is accessed.

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-design-center.adoc:
```diff
@@ -0,0 +1,92 @@
+. Click “+” next to the trigger card.
+. In *Select a component*, search for and select the connector name.
+. Select an operation for the connector.
+. Enter the required values in the *General* tab.
```
Again, please don't make this doc so generic that no actual fields are described. If you cover the info in later sections, provide links to those sections.

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-design-center.adoc:
```diff
@@ -0,0 +1,92 @@
++
+Access to operations on Amazon S3 is further controlled through policies. It is not always possible to validate your credentials before the exact operation for which you have access completes. This can vary based on the bucket name and other parameters. For example, the test connection can fail if your credentials have a restricted policy.
+
+
```
Need a transition here. Users don't see the Connection Type until their 3rd screen: Trigger Card > Set up > Connection Type. Just launching into discussing Connection Types is too hard to follow. Remember too that Design Center users are considered starting users.

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-design-center.adoc:
```diff
@@ -0,0 +1,92 @@
+
+=== Basic Connection Type
+
+Complete the following fields for the *Basic* connection type:
```
Say why one uses the Basic connection type.

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-design-center.adoc:
```diff
@@ -0,0 +1,92 @@
+|===
+
+=== Role Connection Type
+
```
Say why to use this connection type.

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-design-center.adoc:
```diff
@@ -0,0 +1,92 @@
+|===
+|Parameter |Description
+|Configuration Name |Name used to reference the connector instance.
+|Access Key |Alphanumeric text string that uniquely identifies the user who owns the AWS account.
```
Say where users can get the access key

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-design-center.adoc:
```diff
@@ -0,0 +1,92 @@
+|Parameter |Description
+|Configuration Name |Name used to reference the connector instance.
+|Access Key |Alphanumeric text string that uniquely identifies the user who owns the AWS account.
+|Secret Key |Key that plays the role of a password.
```
Say where users can locate the secret key.

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-design-center.adoc:
```diff
@@ -0,0 +1,92 @@
+=== Role Connection Type
+
+Complete the following fields for the *Role* connection type:
+
```
This table is too vague about how a user gets the info. The UI provides more info than this doc.

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-examples.adoc:
```diff
@@ -0,0 +1,155 @@
+image::amazon/amazon-s3-02.png[Studio 7 Visual Studio Icon Flow]
+
+. Create a new Mule project in Studio.
+. Drag an HTTP connector into the canvas, and select it to open the properties editor.
```
This is a Mule 3 statement -- in Studio 7, you drag a connector operation to the canvas.

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-examples.adoc:
```diff
@@ -0,0 +1,155 @@
+|Display Name |Enter a name for the connector instance. In this example, the
+HTTP connector is named Get MuleSoft logo.
+|Configuration |Click the green plus symbol and create a new configuration with the *Host* as `developer.mulesoft.com`, *Protocol+ as HTTPS, and *Port* as 80.
+|URL or Path |Set to ``/sites/all/themes/muletheme/images/mulesoft_dev_logo_v2.svg`.
```
Typo: Shouldn't have 2 tick marks -- should be 1.

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-examples.adoc:
```diff
@@ -0,0 +1,155 @@
+|Field |Value
+|Display Name |Enter a name for the connector instance. In this example, the
+HTTP connector is named Get MuleSoft logo.
+|Configuration |Click the green plus symbol and create a new configuration with the *Host* as `developer.mulesoft.com`, *Protocol+ as HTTPS, and *Port* as 80.
```
Typo: The plus sign should be an asterisk to bold Protocol.

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-studio.adoc:
```diff
@@ -0,0 +1,78 @@
+
+== Install Amazon S3
+
+. In Anypoint Studio, click the Exchange icon in the Studio task bar.
```
Please use the latest steps from the template:

. In Studio, create a Mule Project.
. Click the Exchange *(X)* icon in the Studio task bar.
. In Exchange, click *Login* and supply your Anypoint Platform username and password.
. In Exchange, select *All assets* and search for "<connector>".
. Select <connector>, click *Add to project*.
. Follow the prompts to install the connector.

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-studio.adoc:
```diff
@@ -0,0 +1,78 @@
+. In Exchange, click *Login* and supply your Anypoint Platform username and password.
+. Search for the Amazon S3 connector, click on the connector, and click *Add to Project*.
+. Follow the prompts.
+
```
Provide a note for what operations can be input sources for this connector.

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-studio.adoc:
```diff
@@ -0,0 +1,78 @@
+|===
+|Parameter |Description
+|Name |Name used to reference the connector instance.
+|Access Key |Alphanumeric text string that uniquely identifies the user who owns the AWS account.
```
Need to say where to get these (they come from Amazon), but at least say that, and also check out Amazon to see how much work it is to find those values.

dhirsch1 replied:
AWS is Amazon Web Services - I don't see the need to further qualify that this is AWS is an Amazon offering.

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-studio.adoc:
```diff
@@ -0,0 +1,78 @@
+
+The https://github.com/minio/[Minio Project] is one example of Amazon S3 storage. If you configure this storage locally you might need to set the S3 Compatible Storage URL to `+http://127.0.0.1:9000+`.
+
+== Configure an Input Source
```
This section should be near the top of this doc.

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-xml-maven.adoc:
```diff
@@ -0,0 +1,72 @@
+
+== Add a POM File Dependency
+
+The `pom.xml` Maven file generated by Anypoint Studio contains dependencies for a Mule app. When you code a Mule app manually, include this XML snippet in your `pom.xml` file to enable access to this connector. The Mule runtime engine translates the `LATEST` dependency keyword to the current version.
```
Found from the devs that LATEST doesn't exist. Use RELEASE instead.

george-hoffer commented on modules/ROOT/pages/amazon/amazon-s3-connector-xml-maven.adoc:
```diff
@@ -0,0 +1,72 @@
+<dependency>
+  <groupId>com.mulesoft.connectors</groupId>
+  <artifactId>mule-amazon-s3-connector</artifactId>
+  <version>LATEST</version>
```
Use RELEASE

dhirsch1 committed:
Adding some of George's edits
Adding some of George's editsY

dhirsch1 committed:
Tweaking the changes
Tweaking the changes

dhirsch1 committed:
Committing another try at getting the configure a trigger list formatted correctly.
Committing another try at getting the configure a trigger list formatted correctly.

dhirsch1 committed:
Fixed list in Design Center
Fixed list in Design Center

dhirsch1 committed:
Refixed list in Design Center
Refixed list in Design Center

dhirsch1 committed:
Commit before pushing to beta branch
Commit before pushing to beta branch

dhirsch1 committed:
One more correction
One more correction
